### PR TITLE
Climate

### PR DESCRIPTION
**Pull request by Kurfursten**
forecast
New climate panel for admins

Everything tested

🆑 
* rscadd: Completely overhauled the weather system. You can now see a forecast of weather at climate consoles. Blizzard event still exists as an unforecasted blizzard.
* tweak: Omega Blizzard has been renamed to Dark Season and only lasts 2 hours now. It has a special entry on climate consoles.
* tweak: On Snaxi, you now arrive at the shuttle in the space outpost if you latejoin.
* tweak: If there is a blizzard, you can no longer use the shuttle between the outpost and main station on Snaxi.
* rscadd: The emergency teleportation hub on Snaxi outpost is finally repaired! When a blizzard is underway it will take you down to the surface since shuttles are offline. It's a one way trip.

### CODE REVIEW

**Comment by Kurfursten** [comment]
figured it out... if you don't override process it gets removed from machines

**Comment by Kurfursten** [comment]
fixed a very small bug where on a map without climate you would get thousands of warnings a minute as an admin

File: code/datums/climate.dm [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	for(var/turf/unsimulated/floor/snow/tile in global_snowtiles)
+		if(i == tile_interval)
+			tile.change_snowballs(snowfall_rate[1],snowfall_rate[2])
+			if(tile.snowprint_parent)
+				tile.snowprint_parent.ClearSnowprints()
+			i = 1
+		else
+			i++
+
+var/list/snowstorm_ambience = list('sound/misc/snowstorm/snowfall_calm.ogg','sound/misc/snowstorm/snowfall_average.ogg','sound/misc/snowstorm/snowfall_hard.ogg','sound/misc/snowstorm/snowfall_blizzard.ogg')
```
Comment (N3X15) [inline code/datums/climate.dm:138]: Does this changeset fix the issue where you would hear snow even after even after it ended, or even after you moved into another area?

File: code/datums/climate.dm [hunk 2]
```diff
@@ -0,0 +1,209 @@
+/********************************************
+*                IN THIS FILE               *
+*          Climate Datum, Weather Datum     *
+*                                           *
+********************************************/
+
+#define PREDICTION_MINIMUM 4
+#define PREDICTION_MAXIMUM 10
```
Comment (N3X15) [inline code/datums/climate.dm]: Document what these mean so when you or someone else come back to fix it ten years from now, you'll know what the fuck these do.

**This is the only one you really need to fix.**

File: code/datums/climate.dm [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	else if(istype(weather,/datum/weather))
+		//We have been given a specific weather datum. It may be modified, so run it no matter what.
+		qdel(current_weather)
+		current_weather = weather
+		current_weather.execute()
+
+	else
+		WARNING("Change weather was called with [weather], neither a weather datum nor a path.")
+
+/datum/climate/snow
```
Comment (N3X15) [inline code/datums/climate.dm]: Arctic may be a more fitting name here. \*shrug\*

Comment (N3X15) [reply code/datums/climate.dm]: Also, it may be a good idea to assign a list of possible weathers to the climate.

**Review by N3X15 — CHANGES_REQUESTED** [review]

File: code/datums/climate.dm (hunk 1 above)
Comment (Kurfursten) [reply code/datums/climate.dm:138]: no, I think the only way to really fix that is going to be to break it into smaller, more manageable sounds and that's for another day and someone who has audacity installed

Comment (N3X15) [reply code/datums/climate.dm:138]: If you assign the sound to a specific channel, you can stop it:

```dm
// I think BYOND has a channel limit but can't recall what it is.
#define CHANNEL_WEATHER 10 // Snow, rain, etc
// Snow sounds
usr << sound('sound/weather/snow_average.ogg', channel=CHANNEL_WEATHER)
// Stop snow sounds
usr << sound(null, channel=CHANNEL_WEATHER)

File: code/datums/climate.dm (hunk 3 above)
Comment (Kurfursten) [reply code/datums/climate.dm]: I'll take your name suggestion

Until there are new climates, I'm going to hold off on the other thing though. Currently, one weather has possibilities of others it leads into. Blacklisting may prove useful in a future where we do have alt climates though. We just don't have any reason for that to exist yet.

**Review by N3X15 — APPROVED** [review]

File: code/datums/climate.dm (hunk 1 above)
Comment (Kurfursten) [reply code/datums/climate.dm:138]: yes, it's already on the weather channel, unfortunately as far as I can tell, nowhere does it call null and worse it only plays the (long) soundfile to you if you happened to be on snow when it updated

basically it's glasscode and needs to be rooted up in its own PR with a solid rewrite

**Pull request merged**